File: localNotification/localNotification/ViewController.m
//
//  ViewController.m
//  localNotification
//
//  Created by Vidyashri on 11/09/1941 Saka.
//  Copyright © 1941 CG. All rights reserved.
//

#import "ViewController.h"
#import <UserNotifications/UserNotifications.h>  

@interface ViewController ()

@end

bool isnotificationAccess;

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    isnotificationAccess = false;
    // Do any additional setup after loading the view, typically from a nib.
    
    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
    UNAuthorizationOptions options = UNAuthorizationOptionAlert +UNAuthorizationOptionSound;
    [center requestAuthorizationWithOptions:options completionHandler:^(BOOL granted, NSError * _Nullable error) {
        
        isnotificationAccess = granted;
        
    }];
}

- (IBAction)showNotification:(id)sender {
    if (isnotificationAccess) {
        UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
        UNMutableNotificationContent *content = [[UNMutableNotificationContent alloc] init];
        
        content.title = @"Notification Title";
        
        content.subtitle = @"Notification Subtitle";
        
        content.body = @"This is Obj-C Local Notification";
        
        content.sound = [UNNotificationSound defaultSound];
        
        UNTimeIntervalNotificationTrigger *trigger = [UNTimeIntervalNotificationTrigger triggerWithTimeInterval:10      repeats:NO];
        
        UNNotificationRequest *request = [UNNotificationRequest requestWithIdentifier:@"UYLLocalNotification" content:content      trigger:trigger];
        
        // add notification for current notification centre
        
        [center addNotificationRequest:request withCompletionHandler:nil];
    }
}

@end

File: Animation/Animation/ViewController.swift
//
//  ViewController.swift
//  Animation
//
//  Created by Vidyashri on 03/10/1941 Saka.
//  Copyright © 1941 CG. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var topView: UIView!
    @IBOutlet weak var bottomView: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func bottomButtonpressed(_ sender: UIButton) {
        buttonAnimationAction()
    }
    
    func buttonAnimationAction() {
        UIView.animate(withDuration: 2, animations: {
            self.topView.backgroundColor = .brown
            self.topView.frame.size.width += 10
            self.topView.frame.size.height += 10
        }) { _ in
            UIView.animate(withDuration: 1, delay: 0.25, options: [.autoreverse, .curveEaseOut], animations: {
                self.topView.frame.origin.y -= 20
            }, completion: { (finished: Bool) in
                print("This animation is complete!! :)")
                
            })
        }
    }
    
}

